### DISCUSSION TITLE
No response from ATtiny214 with resistor-only FT232RL UPDI setup

### DISCUSSION THREAD

**Discussion by tylercrumpton** [SpenceKonde/megaTinyCore#724] [Q&A] [answered]
Heya!

I've been struggling to get any sort of response from my ATtiny214 board using the resistor-only version of the UPDI configuration. Can I get a double check on my setup to make sure I'm not missing anything?

- I've got a 1uF ceramic capacitor between VCC and GND as close to the ATtiny214 as possible.
- My FTDI FT232RL board appears to have no in-series resistors on the TXD or RXD pins.
  - The board does have TX and RX activity LEDs, but those are connected to the dedicated pins on the FT232RL
- 4.7k resistor between TX and RX. Multimeter pegs that resistor at 4.64k
- RX on FT232RL connected to UPDI pin on ATtiny214
- 5V on FT232RL connected to VCC on ATtiny214
- GND on FT232RL connected to GND on ATtiny214
- Programmer configured for SLOW speed 57600 baud

Board settings:
<img width="566" alt="image" src="https://user-images.githubusercontent.com/1317406/173009083-87806fbd-a03b-4685-8676-0da33eba3053.png">

Signal on UPDI pin on oscilloscope after clicking "Upload" and triggering on the first long low-signal:
![image](https://user-images.githubusercontent.com/1317406/173008680-6905853f-d904-4348-b24b-149d6d7da685.jpeg)

(From what I can tell, I've got a pretty solid, clean signal going from 5V down to 0.

Arduino version: 1.8.19
OS: macOS Monterey 12.3
megaTinyCore board definition version: 2.5.11

This is my first experience with UPDI and this family of ATtiny MCUs. If anything sticks out as missing, I'd really appreciate the help! I didn't have any Schottky diodes on hand to try, but I'm not against putting in an order for that, and probably will use that long-term if it will help.

**Comment by hmeijdam** [comment]
Have you tried the "latency timer" change from 16 to 1 ms as [described here](https://github.com/SpenceKonde/AVR-Guidance/blob/master/UPDI/jtag2updi.md)?

**Reply by tylercrumpton** [reply]
I haven't tried that yet! I thought it would just be a performance improvement, but if it could affect success, I'll give that a shot!

**Reply by hmeijdam** [reply]
It was just a guess on my end.

**Reply by tylercrumpton** [reply]
I switched over to a Windows machine, and dropped the latency to 1ms, and the timing does look like it was adjusted to match the 1ms latency, but I'm still not getting a response from the ATtiny214. Thanks for the suggestion though! It's looking like it might be something wrong with my wiring, understanding, or my FTDI adapter.

**Reply by hmeijdam** [reply]
The beginning of your signal at least looks like the 0x55 sync signal (page 463 datasheet). The only time I observed a non responsive UPDI, is after[ I disabled it to test the feature of restoring the pin to UPDI](https://github.com/SpenceKonde/megaTinyCore/discussions/450) through a 12 volt pulse.

**Reply by tylercrumpton** [reply]
Hm, these shouldn't have the UPDI disabled or anything, since they're fresh off the reel from Mouser. But if I'm still seeing nothing, I may try to zap them with that HV pulse!

**Comment by mosqu-ito** [comment]
What are those two LEDs telling you?  Tx only, no Rx?
I'd be happier if you had a 104 cap and maybe a 103 too
but that's probably not enough to explain no hint of a reply.
Is that 3.5V high and 1.0V low I'm seeing on the screen?
Try a 10K pullup between UPDI and 5V.  If that doesn't work
try giving Tx a firmer grip (less than 4.7K).  But yeah, the
situation really calls for a little diode with a low Vf
(a schottky signal diode).

**Reply by tylercrumpton** [reply]
I'll check again, but I believe both LEDs were flashing together. I can definitely hunt down different capacitors to try too. The scope was showing a high of 5V and the low of just above 0V. I did try swapping the FTDI voltage over to 3v3, and was seeing the same thing (except the high level was 3.3V obviously). I'll play with adding a pull-up and weakening the 4.7k next! I'll also go ahead and order some Schottky diodes and probably grab some CH340-based adapters while I'm at it. Thanks!

**Comment by mosqu-ito** [comment, marked as answer]
Don't forget to check the simple things.  Make sure you really are getting power onto Vdd and Gnd (far left, Gnd up top). Make sure you've got a solid connection to the UPDI pin and double check that it really is the UPDI pin you're on (not pin 12, 3, or 5).

**Reply by tylercrumpton** [reply]
*sighs* Well, this comment cause me to go and solder up actual pin headers to my board and try again. It worked flawlessly, so either I had something mis-wired, or (more likely) I just never had a solid connection before. Thanks for the push to go eliminate those variables! I feel a little silly now 😅
